src/main/java/org/springframework/data/redis/core/DefaultHashOperations.java

```diff
@@ -141,16 +143,22 @@ public void putAll(K key, Map<? extends HK, ? extends HV> m) {
 
 		byte[] rawKey = rawKey(key);
 
-		Map<byte[], byte[]> hashes = new LinkedHashMap<>(m.size());
+		int size = Math.min(RedisTemplate.REDIS_MAX_ARGS / 2 - 1, m.size());
```

Comment:
We can't change a single anticipated `HMSET` to multiple ones as we're losing atomicity. That's rather something to be handled in the application code.

Reply:
Good point. I didn't think about the atomicity. Removed it again and adapted the test accordingly.